src/util.js

```diff
@@ -107,3 +107,55 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+export function updateContainers() {
+    let newContainers = {};
+    let newContainersStats = {};
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    let newContainersMeta = reply.containers;
+                    let inspectRet = newContainersMeta.map((container) => {
+                        return new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                    .then(reply => resolve(JSON.parse(reply.container)))
+                                    .catch(ex => reject(new Error("Failed to do InspectContainer call:", ex, JSON.stringify(ex))));
+                        });
+                    });
```

Comment:
Note that wrapping `varlinkCall` into a `Promise` doesn't give you anything here. `inspectRet` can just directly be a list of the `varlinkCall` promises. This would further simplifiy the code.

It's also worthwhile to not "destroy" the original exception into a flag string this early -- at some point you want to handle individual exceptions differently in the caller (see  issue #14), so just propagate it as-is to the caller, and let them decide what to do with them (for now, it's ok to just print them).

Also notice that a three-argument Error with (string, exception, string) is rather weird.